### PR TITLE
PWX-29061: Add NearSync related fields

### PR DESCRIPTION
**Pull request by adikulkarni0**
Signed-off-by: Adi Kulkarni <akulkarni@purestorage.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #
PWX-29061
**Special notes for your reviewer**:

### CODE REVIEW

**Review by chiachunlin — APPROVED** [review]

File: api/api.go [hunk 1]
```diff
@@ -511,6 +513,8 @@ type CloudBackupStatusRequest struct {
 	// ignored. This could be GroupCloudBackupId too, and in that case multiple
 	// statuses belonging to the groupCloudBackupID is returned.
 	ID string
+	// Indicates if this is for nearsync
+	NearSyncBackupRestore bool
```
Comment (adityadani) [inline api/api.go:517]: Should we rename this to just `NearSync`? The CloudBackupStatusRequest is used for both backups and restores, and this field does not need to clarify it even further

File: api/api.proto [hunk 2]
```diff
@@ -698,6 +704,10 @@ message VolumeSpec {
   bool winshare = 49;
   // Filesystem create options to be honored.
   string fa_create_options = 50;
+  // NearSync specifies the volume has a near-sync replica
+  bool near_sync = 51;
+  // NearSyncReplicationStrategy is replication startegy for near sync volumes
```
Comment (adityadani) [inline api/api.proto:709]: nit: strategy

File: api/api.proto [hunk 3]
```diff
@@ -771,6 +781,8 @@ message VolumeSpecUpdate {
   oneof readahead_opt { bool readahead  = 41; }
   // winshare is true if this volume can be accessed from windows pods.
   oneof winshare_opt { bool winshare = 42; }
+  // NearSyncReplicationStrategy is replication startegy for near sync volumes
```
Comment (adityadani) [inline api/api.proto:784]: nit: strategy

File: api/api.proto [hunk 4]
```diff
@@ -4464,6 +4476,8 @@ message SdkCloudBackupCreateRequest {
   // DeleteLocal indicates if local snap created for backup must be deleted after
   // the backup is complete
   bool delete_local = 7;
+  // Indicates if this is a backup restore for nearsync
+  bool near_sync_backup_restore = 8;
```
Comment (adityadani) [inline api/api.proto:4480]: Why do we need `restore` as part of this Backup create request? Are we trying to do a restore as part of the same backup request?

Comment (adityadani) [reply api/api.proto:4480]: So looks like we are overloading the BackupCreate request to perform a near-sync migration. Should we introduce a new API `CloudMigrateStart()` instead?

File: api/server/sdk/cloud_backup.go [hunk 5]
```diff
@@ -84,7 +85,13 @@ func (s *CloudBackupServer) Create(
 		if _, ok := err.(*api_err.ErrCloudBackupServerBusy); ok {
 			return nil, status.Errorf(codes.Unavailable, err.Error())
 		}
-		return nil, status.Errorf(codes.Internal, "Failed to create backup: %v", err)
+
```
Comment (adityadani) [inline api/server/sdk/cloud_backup.go:88]: Move this to line 82 and fix the return error messages on line 83

**Pull request merged**